File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/csharp/Empresa/Gerente.cs
// Gerente.cs
using System;
using System.Collections.Generic;

namespace Empresa;

public class Gerente : Funcionario
{
    private readonly List<Funcionario> _equipe = new();

    public double BonusGestao { get; set; }

    public Gerente(string nome, string cpf, double salarioBase, Departamento departamento,
                DateTime dataAdmissao, double bonusGestao)
        : base(nome, cpf, salarioBase, departamento, dataAdmissao)
    {
        BonusGestao = bonusGestao;
    }

    public void AdicionarSubordinado(Funcionario f)
    {
        if (f != null && !_equipe.Contains(f))
        {
            _equipe.Add(f);
        }
    }

    public void RemoverSubordinado(Funcionario f) => _equipe.Remove(f);
    public List<Funcionario> GetEquipe() => new(_equipe);

    public override double CalcularBonus()          => SalarioBase * (BonusGestao / 100) + (_equipe.Count * 100);
    public override double CalcularSalarioTotal()   => SalarioBase + CalcularBonus();

    public override string ToString()
    {
        return $"[GER] {base.ToString()} | Bônus Gestão: {BonusGestao}% | Equipe: {_equipe.Count}";
    }
}

File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/dart/empresa/gerente.dart
class Gerente extends Funcionario {
  List<Funcionario> equipe;
  double bonusGestao; 

  Gerente({
    required String nome,
    required String cpf,
    required double salarioBase,
    required Departamento departamento,
    required DateTime dataAdmissao,
    required this.bonusGestao,
  })  : equipe = [],
        super(
          nome: nome,
          cpf: cpf,
          salarioBase: salarioBase,
          departamento: departamento,
          dataAdmissao: dataAdmissao,
        );

  void adicionarSubordinado(Funcionario f) {
    if (!equipe.contains(f)) {
      equipe.add(f);
    }
  }

  void removerSubordinado(Funcionario f) {
    equipe.remove(f);
  }

  @override
  double calcularBonus() {
    return salarioBase * (bonusGestao / 100) + (equipe.length * 100);
  }

  @override
  double calcularSalarioTotal() {
    return salarioBase + calcularBonus();
  }

  @override
  String toString() {
    return "[GER] ${super.toString()} | Bônus Gestão: $bonusGestao% | Equipe: ${equipe.length}";
  }
}